src/cfg/config.go

```diff
@@ -220,7 +208,17 @@ func (c *Configuration) Init() {
 			}
 			return ot.AccessToken, nil
 		}
-	} else if c.Token != "" {
+	} else if len(c.TokenFilePath) > 1 {
+		// In the case of Kubernetes, the token file can be updated, so this reads it from the file every time.
+		c.tokenFunc = func() (string, error) {
+			tokenBytes, err := ioutil.ReadFile(c.TokenFilePath)
+			if err != nil {
+				return "", err
+			}
+			return string(tokenBytes), nil
+		}
+	} else {
+		c.Token = strings.TrimSuffix(util.AssignString(c.Token, os.Getenv("PulsarToken")), "\n")
```

Comment:
@zzzming - it looks like the OAuth2 implementation might have introduced a minor regression in usability. By always setting this here, it lets users set the token in the main config.json. Let me know if you have any concerns.